azurelinuxagent/ga/exthandlers.py

```diff
@@ -232,6 +232,17 @@ def get_exthandlers_handler(protocol):
     return ExtHandlersHandler(protocol)
 
 
+def list_agent_lib_directory(skip_agent_package=True):
```

Comment:
suggestion: this function doesn't seem to have a well defined functionality (doesn't seem very useful by itself). You are using it in conjunction with get_ext_handler_instance_from_path_if_valid (the "if_valid" is also kind of funny).

why don't you combine both into a single function to get the extension handlers on disk? (in my opinion, such function would be more useful that the other 2).

Reply:
The main reason why I separated out the 2 is because I didnt want to loop through the directory again for checking and removing the orphaned packages in the `_cleanup_outdated_handlers` functions, especially since we call this every 3 secs. 

```python
            if os.path.isfile(path) and not os.path.isdir(path[0:-len(HANDLER_PKG_EXT)]):
                if not re.match(_HANDLER_PKG_PATTERN, item):
                    continue
                pkgs.append(path)
```

If you feel strongly about this though I can combine the 2 functions and then add another loop for the orphaned packages.

Reply:
>(the "if_valid" is also kind of funny).

This basically meant return ext-handler-instance if its a valid handler, but I'll just rename it to something simpler.

Reply:
it was just as suggestion, stroke me as a little odd, but far from "feeling strongly about it" :) ok as is

Reply:
haha fair enough, thanks for the comment :)